### ISSUE TITLE
Potential double averaging of KL Loss in VGAE

### ISSUE THREAD

**Issue by joeybose** [rusty1s/pytorch_geometric#910]
Hi, I'm trying to understand what may be a potential error in the VGAE example. Specifically, it seems the KL loss is averaged twice. For example in model.kl_loss() https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/models/autoencoder.py#L231 there is a torch.mean() which averages over number of nodes. However, in the example code there is another average (1/ data.num_nodes) * model.kl_loss() https://github.com/rusty1s/pytorch_geometric/blob/master/examples/autoencoder.py#L60. I may of course be wrong, but it does seem like a double averaging?

**Comment by rusty1s** [comment]
This is taken from the original VGAE implementation by Thomas, see [here](https://github.com/tkipf/gae/blob/master/gae/optimizer.py#L33). Thomas explains his intuition [here](https://github.com/tkipf/gae/issues/20).

**Comment by joeybose** [comment]
Yes, it seems this is the case. I agree with Thomas in that this could be seen as a beta-VAE and empirically I've seen that you need both the averaging otherwise training is unstable. It is an interesting issue that warrants further investigation as it doesn't fit the canonical VAE framework but that could be down to the messiness of graph data. Regardless, thanks for the pointers and I'll close this issue now.
